src/neo-vm/ExecutionEngine.cs

```diff
@@ -1189,11 +1322,22 @@ internal protected void ExecuteNext()
                 {
                     Instruction instruction = CurrentContext.CurrentInstruction;
                     if (!PreExecuteInstruction() || !ExecuteInstruction() || !PostExecuteInstruction(instruction))
+                    {
                         State = VMState.FAULT;
+                        FaultState.Exception = new InvalidOperationException("OPCode Fault:" + instruction.OpCode.ToString());
```

Comment:
Keep it? It's for debugger, which'll know what exception happen.

Reply:
If it is only for debugging purpose, you can just throw the exception directly, there is no need to store it.

Reply:
Since the exception was catched, debugger can't catch it again.

https://github.com/neo-project/neo-vm/blob/288caa6850d914b520cb9c310738d2c9873145d8/src/neo-vm/ExecutionEngine.cs#L1188-L1196

Reply:
If you store it in `FaultState.Exception`, the debugger can catch it? They are the same.

Reply:
Yes, `FaultState` is public, neo-devpack-dotnet can access it in TestEngine class.

Reply:
It can't be caught by the contract. This is not a function of the virtual machine. So we have to remove it.

Reply:
Done. 

Debuger will not be able to detect error messages when contract execution fails. The direct user of VM is actually the compiler, the debugger.